Sources/Unbox.swift

```diff
@@ -107,6 +107,30 @@ public func unbox<T: UnboxableWithContext>(data: Data, context: T.UnboxContext,
     return try data.unbox(context: context, allowInvalidElements: allowInvalidElements)
 }
 
+/// Unbox binary data into a dictionary of type `[String: T]`. Throws `UnboxError`.
+public func unbox<T: Unboxable>(data: Data) throws -> [String: T] {
+    let json = try JSONSerialization.jsonObject(with: data, options: []) as? [String: [String: Any]]
+    var mappedDictionary = [String: T]()
+    try json?.forEach  { key, value in
```

Comment:
There is no trivial way to map to dictionary in Swift which wouldn't require the chaining of `map` to `reduce` and incur a performance penalty (2 times traversal of an arrary).

I think the mutable state shouldn't be an issue since it's' function-local.